apm-lambda-extension/extension/apm_server.go

```diff
@@ -101,89 +152,81 @@ func PostToApmServer(client *http.Client, agentData AgentData, config *extension
-//
-// This function is public for use in tests.
-func IsTransportStatusHealthyOrPending() bool {
-	return ApmServerTransportState.Status != Failing
-}
-
 // SetApmServerTransportState takes a state of the APM server transport and updates
 // the current state of the transport. For a change to a failing state, the grace period
 // is calculated and a go routine is started that waits for that period to complete
 // before changing the status to "pending". This would allow a subsequent send attempt
 // to the APM server.
 //
 // This function is public for use in tests.
-func SetApmServerTransportState(status ApmServerTransportStatusType, ctx context.Context) {
+func SetApmServerTransportState(ctx context.Context, transport *ApmServerTransport, status ApmServerTransportStatusType) {
 	switch status {
 	case Healthy:
-		ApmServerTransportState.Lock()
-		ApmServerTransportState.Status = status
-		Log.Debugf("APM Server Transport status set to %s", status)
-		ApmServerTransportState.ReconnectionCount = -1
-		ApmServerTransportState.Unlock()
+		transport.Lock()
+		transport.status = status
+		Log.Debugf("APM server Transport status set to %s", transport.status)
+		transport.reconnectionCount = -1
+		transport.Unlock()
 	case Failing:
-		ApmServerTransportState.Lock()
-		ApmServerTransportState.Status = status
-		Log.Debugf("APM Server Transport status set to %s", status)
-		ApmServerTransportState.ReconnectionCount++
-		ApmServerTransportState.GracePeriodTimer = time.NewTimer(computeGracePeriod())
-		Log.Debugf("Grace period entered, reconnection count : %d", ApmServerTransportState.ReconnectionCount)
+		transport.Lock()
+		transport.status = status
+		Log.Debugf("APM server Transport status set to %s", transport.status)
+		transport.reconnectionCount++
+		transport.gracePeriodTimer = time.NewTimer(computeGracePeriod(transport))
+		Log.Debugf("Grace period entered, reconnection count : %d", transport.reconnectionCount)
 		go func() {
 			select {
-			case <-ApmServerTransportState.GracePeriodTimer.C:
+			case <-transport.gracePeriodTimer.C:
 				Log.Debug("Grace period over - timer timed out")
 			case <-ctx.Done():
 				Log.Debug("Grace period over - context done")
 			}
-			ApmServerTransportState.Status = Pending
-			Log.Debugf("APM Server Transport status set to %s", status)
-			ApmServerTransportState.Unlock()
+			transport.status = Pending
+			Log.Debugf("APM server Transport status set to %s", transport.status)
+			transport.Unlock()
 		}()
 	default:
-		Log.Errorf("Cannot set APM Server Transport status to %s", status)
+		Log.Errorf("Cannot set APM server Transport status to %s", status)
 	}
 }
 
 // ComputeGracePeriod https://github.com/elastic/apm/blob/main/specs/agents/transport.md#transport-errors
-func computeGracePeriod() time.Duration {
-	gracePeriodWithoutJitter := math.Pow(math.Min(float64(ApmServerTransportState.ReconnectionCount), 6), 2)
+func computeGracePeriod(transport *ApmServerTransport) time.Duration {
```

Comment:
```suggestion
func (transport *ApmServerTransport) computeGracePeriod() time.Duration {
```

Reply:
Implemented in https://github.com/elastic/apm-aws-lambda/commit/d71ccfb98277e9d98fd558d9fd3be9cf796a6515